experimental/apis/data/v0alpha1/query.go

```diff
@@ -0,0 +1,419 @@
+package v0alpha1
+
+import (
+	"encoding/json"
+	"fmt"
+	"unsafe"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana-plugin-sdk-go/data/converters"
+	"github.com/grafana/grafana-plugin-sdk-go/data/utils/jsoniter"
+	j "github.com/json-iterator/go"
+)
+
+func init() { //nolint:gochecknoinits
+	jsoniter.RegisterTypeEncoder("v0alpha1.DataQuery", &genericQueryCodec{})
+	jsoniter.RegisterTypeDecoder("v0alpha1.DataQuery", &genericQueryCodec{})
+}
+
+type QueryDataRequest struct {
+	// Time range applied to each query (when not included in the query body)
+	TimeRange `json:",inline"`
```

Comment:
I think it would be good to remove this if we can since I don't think it adds any additional functionality - but does add complexity to the Request and handling of the Request.

If there is no additional functionality, and it is just for convenience, then I think it would be better to have Code (e.g. Methods) to set a single on all the queries, rather then have the Type that supports both.